### ISSUE TITLE
pfe-toast

### ISSUE THREAD

**Issue by coreyvickery** [patternfly/patternfly-elements#956]

**Comment by coreyvickery** [comment]
# Alert - Toast

## Usage

### Example
> Toasts are not located inline, they float and stick to the upper right corner of a page or window above content and the grid to alert users of information or actions.
<img width="1678" alt="Screen Shot 2020-06-25 at 11 50 29 AM" src="https://user-images.githubusercontent.com/45767595/85784729-6f539e00-b6dd-11ea-861e-b0efc4804e19.png">

### Desktop
<img width="1440" alt="Toast - Desktop" src="https://user-images.githubusercontent.com/45767595/81442018-cf4bb080-9140-11ea-8a8f-ea3f6e233cec.png">

### Tablet, landscape
<img width="1440" alt="Toast - Tablet, landscape" src="https://user-images.githubusercontent.com/45767595/81442026-d4106480-9140-11ea-88cc-e823bb8841ba.png">

### Tablet, portrait
<img width="1440" alt="Toast - Tablet, portrait" src="https://user-images.githubusercontent.com/45767595/81442039-da064580-9140-11ea-9bf6-05da7abeb11d.png">

**Guidelines**
- Desktop and Tablet Toasts have a fixed width
- Max width is always 4 columns wide or 32% of the layout
- Copy that flows beyond 4 columns or 32% of the layout will break to a second line
- When Toasts stack, the distance between is 8px

### Mobile, landscape
<img width="640" alt="Toast - Mobile, landscape" src="https://user-images.githubusercontent.com/45767595/81442809-4170c500-9142-11ea-98f1-a2f6a92c511b.png">

### Mobile, portrait
<img width="360" alt="Toast - Mobile, portrait" src="https://user-images.githubusercontent.com/45767595/81442819-459ce280-9142-11ea-942a-8bcc72eb229d.png">

**Guidelines**
- The width of Mobile Toasts are 100% of the layout up to 767px
- Copy that flows beyond 100% will break to a second line
- When Toasts stack, the distance between is 8px

## Types

https://xd.adobe.com/spec/3e62e93c-8338-4f31-5040-3b81f0ed5c71-4853/screen/e0be63a2-893e-41eb-96b1-ba9bee4ede0d/Toast

## Styles

### Success
<img width="441" alt="Toast - Success" src="https://user-images.githubusercontent.com/45767595/81430842-272cec00-912e-11ea-9160-0662a829d5e6.png">

**Styles**
- Top line/Border: $pf-color-green-500 (#3e8635)
- Icon: $pf-color-green-500 (#3e8635)
- Title: RH Text, Medium, 14pt / 21 (1.5), $pf-color-green-600 (#1e4f18)
- Copy: RH Text, Regular, 14pt / 21 (1.5), $pf-color-black-900 (#151515)
- Links: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-400 (#06c)
- Close button: $pf-color-black-600 (#6a6e73)
- Background: $pf-color-green-50 (#f3faf2)

### Danger
<img width="441" alt="Toast - Danger" src="https://user-images.githubusercontent.com/45767595/81430867-32801780-912e-11ea-9021-ba70424e1871.png">

**Styles**
- Top line/Border: $pf-color-red-100 (#c9190b)
- Icon: $pf-color-red-100 (#c9190b)
- Title: RH Text, Medium, 14pt / 21 (1.5), $pf-color-red-300 (#7d1007)
- Copy: RH Text, Regular, 14pt / 21 (1.5), $pf-color-black-900 (#151515)
- Links: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-400 (#06c)
- Close button: $pf-color-black-600 (#6a6e73)
- Background: $pf-color-red-50 (#faeae8)

### Warning
<img width="441" alt="Toast - Warning" src="https://user-images.githubusercontent.com/45767595/81430897-4166ca00-912e-11ea-8975-14d6e555ace8.png">

**Styles**
- Top line/Border: $pf-color-gold-400 (#f0ab00)
- Icon: $pf-color-gold-400 (#f0ab00)
- Title: RH Text, Medium, 14pt / 21 (1.5), $pf-color-gold-600 (#795600)
- Copy: RH Text, Regular, 14pt / 21 (1.5), $pf-color-black-900 (#151515)
- Links: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-400 (#06c)
- Close button: $pf-color-black-600 (#6a6e73)
- Background: $pf-color-gold-50 (#fdf7e7)

### Default
<img width="441" alt="Toast - Default" src="https://user-images.githubusercontent.com/45767595/81430915-475cab00-912e-11ea-88d5-27b85ec373b9.png">

**Styles**
- Top line/Border: $pf-color-cyan-300 (#009596)
- Icon: $pf-color-cyan-300 (#009596)
- Title: RH Text, Medium, 14pt / 21 (1.5), $pf-color-cyan-500 (#003737)
- Copy: RH Text, Regular, 14pt / 21 (1.5), $pf-color-black-900 (#151515)
- Links: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-400 (#06c)
- Close button: $pf-color-black-600 (#6a6e73)
- Background: $pf-color-cyan-50 (#f2f9f9)

### Info
<img width="441" alt="Toast - Info" src="https://user-images.githubusercontent.com/45767595/81430924-4c215f00-912e-11ea-9132-edeebb42ef71.png">

**Styles**
- Top line/Border: $pf-color-blue-300 (#2b9af3)
- Icon: $pf-color-blue-300 (#2b9af3)
- Title: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-600 (#004368)
- Copy: RH Text, Regular, 14pt / 21 (1.5), $pf-color-black-900 (#151515)
- Links: RH Text, Medium, 14pt / 21 (1.5), $pf-color-blue-400 (#06c)
- Close button: $pf-color-black-600 (#6a6e73)
- Background: $pf-color-blue-50 (#e7f1fa)

## Spacing

![Toasts-Spacing](https://user-images.githubusercontent.com/45767595/81432845-a1ab3b00-9131-11ea-978c-4e91efab5adb.gif)

**Spacing legend**
https://www.patternfly.org/v4/design-guidelines/styles/spacers

**Comment by coreyvickery** [comment]
Alert - Inline story is below.

https://github.com/patternfly/patternfly-elements/issues/215

**Comment by starryeyez024** [comment]
@coreyvickery  do toasts always have a color, and is the default blue? currently the toast has no color, so I just wanted to confirm. Thanks!

**Comment by coreyvickery** [comment]
@starryeyez024 Yes, `Toast` alerts always have a color bar at the top and the `Default` style can be used as the default.

**Comment by castastrophe** [comment]
@starryeyez024 Should we tackle this as a variant of pfe-toast or do a pfe-alert variant that displays using toast underneath?

**Comment by starryeyez024** [comment]
@castastrophe  My 2 cents at a high-level is that we should make this a design option within pfe-toast, either by making the styles from pfe-alert a mixin, or by leveraging pfe-toast within the shadow dom 👍 

Including the alert demo link here that @heyMP  is working on:
https://deploy-preview-1406--patternfly-elements.netlify.app/elements/pfe-alert/demo/
